src/gosling-track/gosling-track.ts

```diff
@@ -1014,52 +1061,182 @@ function GoslingTrack(HGC: any, ...args: any[]): any {
+                colorToHex(stroke),
+                strokeOpacity, // alpha
+                0.5 // alignment of the line to draw, (0 = inner, 0.5 = middle, 1 = outter)
+            );
+            g.beginFill(colorToHex(color), color === 'none' ? 0 : fillOpacity);
+
+            marks.forEach(d => {
+                if (d.type === 'point') {
+                    const [x, y, r = 3] = d.polygon;
+                    g.drawCircle(x, y, r);
+                } else if (d.type === 'line') {
+                    g.moveTo(d.polygon[0], d.polygon[1]);
+                    flatArrayToPairArray(d.polygon).map(d => g.lineTo(d[0], d[1]));
+                } else {
+                    g.drawPolygon(d.polygon);
+                }
+            });
+        }
+
+        onRangeBrush(range: [number, number] | null, skipApiTrigger = false) {
+            this.pMouseSelection.clear();
+
+            if (range === null) {
+                // brush just removed
+                if (!skipApiTrigger) {
+                    PubSub.publish('rangeselect', { id: this.viewUid, genomicRange: null, data: [] });
+                }
+                return;
+            }
+
+            const [startX, endX] = range;
+
+            // Collect all gosling track models
+            const models: GoslingTrackModel[] = this.visibleAndFetchedTiles()
+                .map(tile => tile.goslingModels ?? [])
+                .flat();
+
+            // Collect all mouse event data from tiles and overlaid tracks
+            let capturedElements: MouseEventData[] = models
+                .map(model => model.getMouseEventModel().findAllWithinRange(startX, endX, true))
+                .flat();
+
+            // Deselect marks if their siblings are not selected.
+            // e.g., if only one exon is selected in a gene, we do not select it.
+            const idField = this.options.spec.experimental?.groupMarksByField;
+            if (capturedElements.length !== 0 && idField) {
+                models.forEach(model => {
+                    const siblings = model.getMouseEventModel().getSiblings(capturedElements, idField);
+                    const siblingIds = Array.from(new Set(siblings.map(d => d.value[idField])));
+                    capturedElements = capturedElements.filter(d => siblingIds.indexOf(d.value[idField]) === -1);
+                });
+            }
+
+            if (capturedElements.length !== 0) {
+                // selection effect graphics
+                const g = this.pMouseSelection;
+                const stroke = this.options.spec.experimental?.selection?.stroke ?? 'black';
+                const strokeWidth = this.options.spec.experimental?.selection?.strokeWidth ?? 1;
+                const strokeOpacity = this.options.spec.experimental?.selection?.strokeOpacity ?? 1;
+                const color = this.options.spec.experimental?.selection?.color ?? 'none';
+                const fillOpacity = this.options.spec.experimental?.selection?.opacity ?? 1;
```

Comment:
and then this can be simplified

```typescript
let options = {
    stroke: 'black', 
    strokeWidth: 1,
    strokeOpacity: 1,
    color: 'none',
    opacity: 1,
    ...this.options.spec.experimental?.selection
};